### PR TITLE
Adding buildings

### PR DESCRIPTION
**Pull request by sschrijver**
- [ ] Code review
- [ ] Comments
- [ ] Functional tests written
- [ ] Functional tests checked by someone other than the author(s)
- [ ] No linting errors
- [ ] Unit tests passes
- [ ] Technical documentation written
- [ ] Technical documentation reviewed

### CODE REVIEW

File: src/logic/entity/static/building/building_entity.cpp [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    state_machine = new StateMachine<BuildingEntity>(this);
+}
+
+BuildingType BuildingEntity::get_building_type() {
+    return building_type;
+}
+//
+//void BuildingEntity::update(float f) {
+//    state_machine->update();
```
Comment (JeroenVinke) [inline src/logic/entity/static/building/building_entity.cpp]: if we're not going to use this anymore, please remove it

File: src/logic/entity/static/building/building_entity.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+     * This will change the texture of a building entity.
+     * The first boolean will determine if the texture should be transparent
+     * The second boolean will determine if the texture needs a red border
+     * The red border will only work if the object is transparent
+     */
+    virtual void set_to_transparent(bool, bool = true) = 0;
+
+    BuildingType get_building_type();
+
+//    virtual void update(float);
```
Comment (JeroenVinke) [inline src/logic/entity/static/building/building_entity.h]: same as above, if we're not going to use it anymore, please remove this line

File: src/logic/world/world.h [hunk 3]
```diff
@@ -14,14 +14,19 @@
 #include "sdl/sdl_render_object.h"
 
 class BSPTree;
+
 class MovingEntity;
 class Player;
 class BaseEntity;
 
+class Player;
```
Comment (JeroenVinke) [inline src/logic/world/world.h]: this is a duplicate forward declaration

File: src/main.cpp [hunk 4]
```diff
@@ -166,136 +170,131 @@ int main(int argc, char **argv) {
...
+//
+//    MouseHandlerWorld *world_panel_slot = new MouseHandlerWorld();
+//
+//    main_panel.set_mouse_callback(world_panel_slot);
+//    main_window.set_mouse_callback(world_panel_slot);
+//
+//
+//    mouse_dispatcher->register_callback(&main_panel, world_panel_slot);
+//    mouse_dispatcher->register_callback(&main_window, world_panel_slot);
+//>>>>>>> 480dbb209051acdf092b580ad99a522136731324
```
Comment (JeroenVinke) [inline src/main.cpp]: it could be confusing to see a merge conflict marker here

File: src/ui/CMakeLists.txt [hunk 5]
```diff
@@ -23,7 +23,7 @@ include_directories(${CMAKE_SOURCE_DIR}/include)
 include_directories(../renderer ../logic)
 
 add_library(cd_ui STATIC
-            ${SOURCE_FILES} )
+            ${SOURCE_FILES} sdl/event/slot/sdl_key_event_slot.cpp sdl/event/slot/sdl_key_event_slot.h sdl/event/sdl_key_event_dispatcher.h sdl/event/sdl_key_event_dispatcher.cpp)
```
Comment (JeroenVinke) [inline src/ui/CMakeLists.txt]: could you make this more readable?

File: src/main.cpp [hunk 6]
```diff
@@ -186,15 +187,23 @@ int main(int argc, char **argv) {
...
-    SDLWindow *main_window = new SDLWindow(main_panel_representation, window, &render_engine, mouse_dispatcher);
+    SDLWindow *main_window = new SDLWindow(main_panel_representation, window, &render_engine, mouse_dispatcher, key_dispatcher);
 
     MouseHandlerWorld *world_panel_slot = new MouseHandlerWorld();
 
     main_panel.set_mouse_callback(world_panel_slot);
 
+    SDL_KeyEventSlot * key_slot = new SDL_KeyEventSlot();
+
+    main_panel.set_key_callback(key_slot);
```
Comment (Rboin) [inline src/main.cpp]: Key callback should be registered to the main_widow instead, since there is only one key handler

File: src/main.cpp [hunk 7]
```diff
@@ -186,15 +187,23 @@ int main(int argc, char **argv) {
...
+
+    key_dispatcher->register_callback(&main_panel, key_slot);
```
Comment (Rboin) [inline src/main.cpp]: key_dispatcher->register_callback(&main_panel, key_slot); -> key_dispatcher->register_callback(main_window, key_slot);

**Pull request merged**